README.md

```diff
@@ -382,23 +382,26 @@ We have a small collection of algorithms, one for every day of the month. Scroll
         ```math
         M = n(n^2+1)/2
         ```
-    - Now the problem is to generate all the possible magic squares if the order of the square is given as input.
-
+    - Now the problem is to generate all the possible magic squares if the order of the square and magic sum is given as input.
+    - `Hint:` It will be helpful if you breakdown the problem. First generate one magic square. Before generating the others, look at the other squares in the example carefully, you will notice something very useful.
 
   - **Example**
+
    <img src="/src/docs/Dec17-Example.png" width="375" height="288"/>
 
   - **Resources**
-    - [Magic Squares #1](http://www.math.wichita.edu/~richardson/mathematics/magic%20squaresorder3magicsquare.html)
-    - [Magic Squares #2](https://mindyourdecisions.com/blog/2015/11/08/how-many-3x3-magic-squares-are-there-sunday-puzzle/)
+    - [Magic Squares #1](https://mindyourdecisions.com/blog/2015/11/08/how-many-3x3-magic-squares-are-there-sunday-puzzle/)
+    - [Magic Square #2](https://en.wikipedia.org/wiki/Magic_square)
 
 
 ### **December 18 - How Secure is your Password?**
   - **Problem**
-    - Brute Force Attack is the most widely known password cracking method. This attack simply tries to use every possible printable ASCII character combination as a password. Though brute forcing can be a little exhaustive, the likelihood of a password being cracked is made certain.
+    - Brute Force Attack is the most widely known password cracking method. This attack simply tries to use every possible ASCII printable characters (character code 32-126) combination as a password. Though brute forcing can be a little exhaustive, the likelihood of a password being cracked is made certain.
```

Comment:
I think that mentioning character codes and everything in particular makes it kinda complicated for the reader. Just say that there's only 72 or whatever number of allowed characters for the password.

Reply:
Plus they aren't actually going to use the characters themselves no? It'll be just calculating the total number of combinations/500,000.

Reply:
Sure thing bruh! I'll change em.

Reply:
The tricky part is that loop(s) are required to find the approx. time needed as the number of letters increases.

Reply:
- Time Taken to brute-force ``h3lL0``
``((10*68*36*8)+(27*94*94*94*94))+((94^4)+(94^3)+(94^2)+94) = 4374.26 seconds``
- That's what I tried to do.
- I can change the problem statement to **time needed as the number of letters increases** if the above is difficult.

Reply:
Uh I'm confused.

Time taken to bruteforce `h31L0` should be time taken for 1 char + 2 chars + 3 chars + 4 chars + 5 chars no?

How are you getting (10*68*36*8)?

| Number of Chars | Calculation | Combinations | Time Taken |
| --- | --- | --- | --- |
| 1 | 92 | 92 | <1s |
| 2 | 92 + 92^2 | 8556 | <1s |
| 3 | 92 + 92^2 + 92^3 | 787244 | 1.57s |
| 4 | 92 + 92^2 + 92^3 + 92^4 | 72426540 | 144.85s |
| 5 | 92 + 92^2 + 92^3 + 92^4 + 92^5 | 6663241772 | 13326.483544s |

Which was your original output right?

Reply:
Yes, but later I realized by that logic even `aaaaa` and `H3Ll0`  would take the same amount of time while Brute-Forcing traverse char-by-char.

Reply:
It's your call. Which one do I have, finding the time required **by the length of the string** or **Number of tries to reach the string divided by Number of passwords that can be tested per second**?

Reply:
While what you're saying is more accurate, it's a bit more complicated to implement. I think we can just use the **length of the string** logic and call it the maximum time taken to crack the password.

Also as an aside, the order of bruteforcing may also vary. I'd venture that the most popular characters are tried first.

Reply:
Aight, I'll go with the **maximum time taken to crack the password** and have the other one as the optional part.

Reply:
Sure.